conf_collectd_plugin.py

```diff
@@ -0,0 +1,375 @@
+      'To enable a server-status page, the following code in quote\n'
+      '"<Location /server-status>\n'
+      '  SetHandler server-status\n'
+      '</Location>"\n'
+      'must be included within a <VirtualHost> block '
+      'for the .conf file of your server.\n')
+
+
+def write_apache_plugin(out):
+    out.write('LoadPlugin "apache"\n')
+    out.write('<Plugin "apache">\n')
+
+    count = 0
+    server_list = []
+
+    apache_plugin_usage()
+    sys.stdout.write(
+        "To check whether the server-status page is working, please visit\n"
+        "\tyour-server-name/server-status\n"
+        "It should look similar to\n"
+        "\tapache.org/server-status\n")
+
+    while(utils.ask("Would you like to add a server to monitor?")):
+        url = utils.get_input(
+            "Please enter the url that contains your " +
+            "server-status (ex: www.apache.org/server_status):")
+        print ""
+        utils.print_step("Checking http response for %s" % url)
+        res = utils.get_command_output('curl -s -i '+url)
+
+        if res is None:
+            ret = INVALID_URL
+        else:
+            ret = check_http_response(res)
+
+        if ret == NOT_AUTH:
+            # skip for this case for now, ask for user/pass
+            sys.stderr.write(
+                "Authorization server status is required, please "
+                "try again.\n")
+        elif ret == NOT_FOUND or ret == INVALID_URL:
+            utils.print_failure()
+            sys.stderr.write(
+                "Invalid url was provided, please try "
+                "again.\n")
+        elif ret == HTTP_OK:
+            utils.print_success()
+            if url in server_list:
+                utils.print_warn("You have already added this server instance.")
+            else:
+                res = check_apache_server_status(res)
+                if res is None:
+                    utils.print_warn(
+                        "The url you have provided "
+                        "does not seem to be the correct server_status "
+                        "page.  Incorrect server-status will not be recorded "
+                        "by collectd.")
+                    utils.ask("Would you like to record this url anyway?", "no")
+                else:
+                    print res
+                    res = utils.ask("Is this the correct status to monitor?")
+                    print ""
+                    if res:
+                        count = count + 1
+                        server_list.append(url)
+                        instance = "apache%d" % count
+                        url_auto = url+"?auto"
+                        out.write('  <Instance "'+instance+'">\n')
+                        out.write('    URL "'+url_auto+'"\n')
+                        out.write('  </Instance>\n')
+
+    out.write('</Plugin>\n')
+    return count
+
+
+def print_log():
+    print "Writing logs to %s." % config.INSTALL_LOG
+
+
+if __name__ == "__main__":
```

Comment:
I _have_ seen python conventions where if a module is main, it runs the tests.  However, I would like the output to be very explicit and clear that it's running a test if that happens, so that the user won't get confused.  (I understand this is an unlikely case)